### PR TITLE
CB-16851 CB should not find out role and resource role CRNs but query…

### PR DESCRIPTION
**Pull request by horadla23**
… them from UMS

See detailed description in the commit message.

### CODE REVIEW

File: auth-connector/src/test/java/com/sequenceiq/cloudbreak/auth/altus/service/RoleCrnGeneratorTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    private RoleCrnGenerator underTest;
+
+    @Test
+    public void testGetRoleCrn() {
+        when(grpcUmsClient.getRoles(any(), any())).thenReturn(Set.of(
+                "crn:altus:iam:us-west-1:altus:role:TestRole1",
+                "crn:altus:iam:us-west-1:altus:role:TestRole2"));
+        Crn testRole1 = ThreadBasedUserCrnProvider.doAs(ACTOR, () -> underTest.getRoleCrn(TEST_ROLE_1));
+        Crn testRole2 = ThreadBasedUserCrnProvider.doAs(ACTOR, () -> underTest.getRoleCrn(TEST_ROLE_2));
+        assertThrows(InternalServerErrorException.class, () -> ThreadBasedUserCrnProvider.doAs(ACTOR, () -> underTest.getRoleCrn("whatever")));
```
Comment (bergerdenes) [inline auth-connector/src/test/java/com/sequenceiq/cloudbreak/auth/altus/service/RoleCrnGeneratorTest.java:46]: I would put the negative test to a separate method

File: auth-connector/src/test/java/com/sequenceiq/cloudbreak/auth/altus/service/RoleCrnGeneratorTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    @Test
+    public void testGetResourceRoleCrn() {
+        when(grpcUmsClient.getResourceRoles(any(), any())).thenReturn(Set.of(
+                "crn:altus:iam:us-west-1:altus:resourceRole:TestRole1",
+                "crn:altus:iam:us-west-1:altus:resourceRole:TestRole2"));
+        Crn testRole1 = ThreadBasedUserCrnProvider.doAs(ACTOR, () -> underTest.getResourceRoleCrn(TEST_ROLE_1));
+        Crn testRole2 = ThreadBasedUserCrnProvider.doAs(ACTOR, () -> underTest.getResourceRoleCrn(TEST_ROLE_2));
+        assertThrows(InternalServerErrorException.class, () ->
```
Comment (bergerdenes) [inline auth-connector/src/test/java/com/sequenceiq/cloudbreak/auth/altus/service/RoleCrnGeneratorTest.java:64]: I would put the negative test to a separate method

**Review by bergerdenes — CHANGES_REQUESTED** [review]

File: auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/UmsClient.java [hunk 3]
```diff
@@ -1156,6 +1151,37 @@ public GetUserSyncStateModelResponse getUserSyncStateModel(
...
     }
 
+    public Set<String> listResourceRoles(String requestId, String accountId) {
+        UserManagementProto.ListResourceRolesResponse response = newStub(requestId).listResourceRoles(UserManagementProto.ListResourceRolesRequest.newBuilder()
+                .setAccountId(accountId)
+                .build());
+        Set<String> resourceRoles = Sets.newHashSet();
+        resourceRoles.addAll(response.getResourceRoleList().stream().map(resourceRole -> resourceRole.getCrn()).collect(Collectors.toSet()));
+        while (response.hasNextPageToken()) {
+            response = newStub(requestId).listResourceRoles(UserManagementProto.ListResourceRolesRequest.newBuilder()
```
Comment (foldik) [inline auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/UmsClient.java:1161]: Please extract the stub into a local variable and not create a new one every time in a loop.

File: auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/UmsClient.java [hunk 4]
```diff
@@ -1156,6 +1151,37 @@ public GetUserSyncStateModelResponse getUserSyncStateModel(
...
+    }
+
+    public Set<String> listRoles(String requestId, String accountId) {
+        UserManagementProto.ListRolesResponse response = newStub(requestId).listRoles(UserManagementProto.ListRolesRequest.newBuilder()
+                .setAccountId(accountId)
+                .build());
+        Set<String> roles = Sets.newHashSet();
+        roles.addAll(response.getRoleList().stream().map(role -> role.getCrn()).collect(Collectors.toSet()));
+        while (response.hasNextPageToken()) {
+            response = newStub(requestId).listRoles(UserManagementProto.ListRolesRequest.newBuilder()
```
Comment (foldik) [inline auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/UmsClient.java:1177]: Extract the stub here too.

File: auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/cache/UmsRolesCache.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+@Service
+public class UmsRolesCache extends AbstractCacheDefinition {
+
+    private static final long MAX_ENTRIES = 1000L;
+
+    /**
+     * @deprecated {@link #ttlMinutes} was replaced by {@link #ttlSeconds} because it was not providing enough flexibility on ttl.
+     */
+    @Value("${altus.ums.roles.cache.ttl:5}")
```
Comment (foldik) [inline auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/cache/UmsRolesCache.java:18]: Please remove the deprecated field and use just `ttlSeconds` with 300 seconds.

**Review by foldik — CHANGES_REQUESTED** [review]
I requested a few changes otherwise it looks good.

**Review by bergerdenes — APPROVED** [review]

**Review by foldik — APPROVED** [review]
LGTM

**Pull request merged**